packages/twenty-server/src/engine/workspace-manager/workspace-sync-metadata/services/workspace-metadata-updater.service.ts

```diff
@@ -230,6 +232,69 @@ export class WorkspaceMetadataUpdaterService {
     };
   }
 
+  async updateIndexMetadata(
+    manager: EntityManager,
+    storage: WorkspaceSyncStorage,
+    originalObjectMetadataCollection: ObjectMetadataEntity[],
+  ): Promise<{
+    createdIndexMetadataCollection: IndexMetadataEntity[];
+  }> {
+    const indexMetadataRepository = manager.getRepository(IndexMetadataEntity);
+
+    const convertIndexMetadataForSaving = (
+      indexMetadata: PartialIndexMetadata,
```

Comment:
I'm not clear with the naming PartialIndexMetadata vs IndexMetadataEntity here too.
I think we should be very clear: IndexMetadataEntity is something ready for typeORM. PartialIndexMetadata could be another data structure that only contains narrower information (but do we need it though?)